lib/model.js

```diff
@@ -1027,19 +1028,21 @@ Model.prototype.dropSchema = function(schema) {
  * @return {this}
  */
 Model.prototype.schema = function(schema, options) { // testhint options:none
-  this.options.schema = schema;
+  var self = Object.create(this);
```

Comment:
Perhaps use `clone` rather than `self` since `self` usually is a reference to the original `this`.